guides/hack/01-getting-started/02-tools.md

```diff
@@ -1,3 +1,47 @@
-**Under Construction**
+## Tools
 
-A placeholder for tools info.
+The core tools are:
+
+- `hh_client`: this is the command line interface for Hack's static analysis; it
+  is needed to verify that a project is valid Hack, and is used to find errors
+  in your programs
+- `hhvm`: this is used to execute your Hack code, and can either be used for
+  CLI (e.g. `hhvm foo.hack`) or as a server, and has
+  [extensive documentation](/hhvm/)
+
+### Editors and IDEs
+
+While you can edit Hack with any text editor, enhanced support is available
+for several.
+
+We primarily recommend using [Visual Studio Code] with the [VSCode-Hack]
+extension; this provides standard IDE-like features such as
+syntax highlighting, go-to-definition, and in-line display of Hack errors.
+
+For Vim users, [vim-hack] provides syntax highlighting and language detection,
+and the [ALE] project provides enhanced support for Hack.
+
+[hack-mode] provides a major mode for Emacs users.
+
+### Dependency Management
+
+Hack dependencies are currently managed using [Composer], which must be executed
+with PHP.  Composer can be thought of as an equivalent to `npm` or `yarn`.
+
+### Other Common Tools
+
+- `hackfmt` is a CLI code formatter included with HHVM and Hack, and is also
+  used by the various editor and IDE integrations
+- [HHAST] is added provides code style linting, and the ability to automatically
+  modify code to adapt to some changes in the language or libraries
+- [hacktest] and [fbexpect] are commonly used together for writing unit tests
+
+[ALE]: https://github.com/w0rp/ale
+[Composer]: https://getcomposer.org
+[HHAST]: https://github.com/hhvm/hhast
+[VSCode-Hack]: https://github.com/slackhq/vscode-hack/
+[Visual Studio Code]: https://code.visualstudio.com
+[fbexpect]: https://github.com/hhvm/fbexpect
+[hack-mode]: https://github.com/hhvm/hack-mode
+[hack-test]: https://github.com/hhvm/hacktest
```

Comment:
I suggest you use inline links rather than putting them at the end. It makes longer files easy to edit without scrolling between the link and its usage.

In any case, I think you want `hacktest` here :P

Reply:
it really hurts readability though, especially with generally sticking to 80col. Nice catch though :)

Reply:
In which case I believe you can put the links immediately after the paragraph, rather than at the end of the file.